backend/src/libs/exceptions/auth-error/auth-error.exception.ts

```diff
@@ -0,0 +1,16 @@
+import { type HTTPCode, HTTPError } from '#libs/packages/http/http.js';
```

Comment:
Please move it to shared, we will use it on frontend too, when we will need to detect AuthError to sign out user

Reply:
now it doesn't let me type status as optional. Can I set the default value for status as unauthorized for example? Or what do you think I should do in this case?

![image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/58943798/cea19807-26a2-41db-8062-9180e7eb546f)

Reply:
For AuthError status is not optional

Reply:
earlier in the discussion, it was suggested to me not to write status when throwing an error https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/pull/39#discussion_r1301564582. Now I provided the default value in the AuthError. If it is not an option, please, help me understand where it should be assigned

Reply:
oh, ok got this 😅 sorry for confusion, let's add statuses to custom errors that extends Http, because we will use them per service not per status code. I think Vova was thinking about the second approach

Reply:
could you check please what I've done and tell me if something should be changed?

Reply:
checked it again, a few comments, everything else looks pretty good 🙌